openfe/setup/ligand_network_planning.py

```diff
@@ -283,46 +279,35 @@ def generate_minimal_redundant_network(
       Minimum Spanning Tree number: the number of minimum spanning trees to
       generate. If two, the second-best edges are included in the returned
       network. If three, the third-best edges are also included, etc.
+    n_processes: int
+
     """
     if isinstance(mappers, AtomMapper):
         mappers = [mappers]
-    mappers = [_hasten_lomap(m, ligands) if isinstance(m, LomapAtomMapper)
-               else m for m in mappers]
-
-    # First create a network with all the proposed mappings (scored)
-    network = generate_maximal_network(ligands, mappers, scorer, progress)
-
-    # Flip network scores so we can use minimal algorithm
-    g2 = nx.MultiGraph()
-    for e1, e2, d in network.graph.edges(data=True):
-        g2.add_edge(e1, e2, weight=-d['score'], object=d['object'])
-
-    # As in .generate_minimal_spanning_network(), use nx to get the minimal
-    # network. But now also remove those edges from the fully-connected
-    # network, then get the minimal network again. Add mappings from all
-    # minimal networks together.
-    mappings = []
-    for _ in range(mst_num):  # can increase range here for more redundancy
-        # get list from generator so that we don't adjust network by calling it:
-        current_best_edges = list(nx.minimum_spanning_edges(g2))
+    mappers = [
+        _hasten_lomap(m, ligands) if isinstance(m, LomapAtomMapper) else m
+        for m in mappers
+    ]
+    nodes = list(ligands)
 
-        g2.remove_edges_from(current_best_edges)
-        for _, _, _, edge_data in current_best_edges:
-            mappings.append(edge_data['object'])
+    # Construct network
+    network_planner = RedundantMinimalSpanningTreeNetworkGenerator(
+        mappers=mappers,
+        scorer=scorer,
+        progress=progress,
+        n_redundancy=mst_num,
+        n_processes=n_processes,
+    )
 
-    redund_network = LigandNetwork(mappings)
-    missing_nodes = set(network.nodes) - set(redund_network.nodes)
-    if missing_nodes:
-        raise RuntimeError("Unable to create edges to some nodes: "
-                           f"{list(missing_nodes)}")
+    network = network_planner.generate_ligand_network(nodes)
 
-    return redund_network
+    return network
 
 
 def generate_network_from_names(
-        ligands: list[SmallMoleculeComponent],
-        mapper: AtomMapper,
-        names: list[tuple[str, str]],
+    ligands: list[SmallMoleculeComponent],
+    mappers: Union[AtomMapper, Iterable[AtomMapper]],
```

Comment:
The argument rename is an API break, we should do our best to avoid this in a user facing method in 1.x.

Reply:
I see your reasoning, and I think it makes a lot of sense.  
still maybe for a 2.x you could mark this change, as it harmonizes the function calls. (or you have totally different solution)